be/src/exec/pipeline/source_operator.h

```diff
@@ -27,5 +27,19 @@ class SourceOperator : public Operator {
     MorselQueue* _morsel_queue;
     ChunkSourcePtr _chunk_source;
 };
+
+class SourceOperatorFactory : public OperatorFactory {
+public:
+    SourceOperatorFactory(int32_t id, int32_t plan_node_id) : OperatorFactory(id, plan_node_id) {}
+    bool is_source() const override { return true; }
+    virtual bool with_morsels() const { return false; }
```

Comment:
could you add some annotations to explain this method, I'm a little confused

Reply:
ok, at present, only ScanOperator need attach morsel queue. other source operators, like ExchangeSource, LocalExchangeSource have no morsel queue to attach.